friday/stages_par.py

```diff
@@ -0,0 +1,383 @@
+    struct = []
+    nodeId = []
+    for node in nodeList:
+        logFile.write("Checking node: " + node + "\n")
+        if node in nodeId:
+            ind = nodeId.index(node)
+            if len(ind):
+                list_files = struct[ind]
+            else:
+                list_files = []
+        logFile.write("Checking node: " + node + '\n')
+        datNode = datlib.get_list_file(project.path + '\\DATA', node)
+        datlib.check_list_files(datNode, logFile)
+    logFile.write("Done! You can now check the file " + project.path + "\\checkDataFolder.log")
+    logFile.close()
+
+    project.close_friday_log(log_file)
+
+    return nodeList
+
+def get_records(project):
+
+    log_file = project.open_friday_log()
+    # Checking if previous library file is existing
+    if len(glob.glob(project.path + '\\nodeLibrary.dat')):
+        records = datlib.read_library_file(project.path + '\\nodeLibrary.dat')
+        records, project.list_nodes = datlib.add_new_nodes(records, project.list_nodes)
+    else:
+        records = [[] for i in range(len(project.list_nodes))]
+
+    # Reading recording file
+    injections = datlib.read_log_file(project.recording_file)
+
+    # Using a pool of worker to read each node folder
+    max_number_processes = multiprocessing.cpu_count()
+    pool = multiprocessing.Pool(max_number_processes - 2)
+    r = []
+    print("\tStarting workers pool: " + str(max_number_processes - 2) + " workers reading files.")
+
+    for i in range(len(project.list_nodes)):
+        #res = pool.map(datlib.read_node_library, )
+        r.append(pool.apply_async(datlib.read_node_library, (project.path, project.list_nodes[i], records, i)))
+    pool.close()
+    pool.join()
+    for i in range(len(r)):
+        tmp = r[i].get()
+        records[i] = tmp[0]
+        project.write_messages_to_friday(log_file, tmp[1])
+
+    datlib.make_library_file(project.list_nodes, records, project.path)
+
+    # Comparing recording file and list of records
+    for injection in injections:
+        for nodeIndex in range(len(project.list_nodes)):
+            for record in records[nodeIndex]:
+                test = datlib.is_node_active(injection, record)
+                if test:
+                    try:
+                        tmp = record.getUtm()
+                        injection.list_nodes.append(record.node_id)
+                        injection.list_gps.append([tmp[0], tmp[1], record.altitude])
+                        if record.node_type == 'C':
+                            injection.list_type.append('C')
+                        else:
+                            injection.list_type.append(record.relay_state)
+                        injection.list_files.append(record.name)
+                    except:
+                        pass
+    project.close_friday_log(log_file)
+
+    return records, injections
+
+def harmonic_vp_analysis_injection(project, injection):
+
+    log_file = project.open_friday_log()
+    sample_freq = 150
```

Comment:
Moving forward we need to avoid hard coding. Depending on the survey conducted, sampling frequency will change.

Reply:
Yeah, those were hard coded for tests, I have to include the sample frequency here, that will be resolved next upload.